Ooui/JsonConvert.cs

```diff
@@ -36,60 +38,44 @@ public static void WriteJsonString (System.IO.TextWriter w, string s)
 
         public static void WriteJsonValue (System.IO.TextWriter w, object value)
         {
-            if (value == null) {
-                w.Write ("null");
-                return;
-            }
-            var s = value as string;
-            if (s != null) {
-                WriteJsonString (w, s);
-                return;
-            }
-
-            var a = value as Array;
-            if (a != null) {
-                w.Write ('[');
-                var head = "";
-                foreach (var o in a) {
-                    w.Write (head);
-                    WriteJsonValue (w, o);
-                    head = ",";
-                }
-                w.Write (']');
-                return;
-            }
-
-            var e = value as EventTarget;
-            if (e != null) {
-                w.Write ('\"');
-                w.Write (e.Id);
-                w.Write ('\"');
-                return;
-            }
-
-            if (value is Color) {
-                WriteJsonString (w, ((Color)value).ToString ());
-                return;
-            }
-
-            var icult = System.Globalization.CultureInfo.InvariantCulture;
-
-            if (value is double) {
-                w.Write (((double)value).ToString (icult));
-                return;
-            }
-
-            if (value is int) {
-                w.Write (((int)value).ToString (icult));
-                return;
-            }
-
-            if (value is float) {
-                w.Write (((float)value).ToString (icult));
-                return;
+            switch(value) {
+                case string s:
+                    WriteJsonString (w, s);
+                    break;
+                case Array a:
+                    w.Write ('[');
+                    var head = "";
+                    foreach (var o in a) {
+                        w.Write (head);
+                        WriteJsonValue (w, o);
+                        head = ",";
+                    }
+                    w.Write (']');
+                    break;
+                case EventTarget e:
+                    w.Write ('\"');
+                    w.Write (e.Id);
+                    w.Write ('\"');
+                    break;
+                case Color c:
+                    WriteJsonString (w, c.ToString());
+                    break;
+                case double d:
+                    w.Write (Invariant ($"{d}"));
```

Comment:
The `Invariant` function has quite a bit of overhead compared to my previous code (several more allocations and several abstract method calls). I prefer the old way since the whole point of this function is to be fast.